### PR TITLE
[64] Allow pseudonym replacement even if assassin is not in event

### PR DESCRIPTION
**Pull request by jsyiek**
Implements #64 

**Problem**: By default, when [PX]/etc. is in a report/headline and is not an assassin explicitly added to an event, the pseudonym is not rendered.

**Solution**: Allow it to be rendered by performing some regex search to figure out which assassins are in the report.

**Testing**: Manual

### CODE REVIEW

**Review by thenewtonium — CHANGES_REQUESTED** [review]

**Comment by thenewtonium** [comment]
Testing this it has occured to me that hiding deleted pseudonyms makes it hard to figure out what the appropriate index for a given pseudonym is when using the [PX_i] format.

Perhaps blank pseudonyms shouldn't be hidden in Assassin -> Update (but still be hidden in Event -> Create and Event -> Update and when rendering [DX])...

File: AU2/plugins/custom_plugins/PageGeneratorPlugin.py [hunk 1]
```diff
@@ -227,6 +232,38 @@ def on_page_generate(self, _) -> List[HTMLComponent]:
...
+
+                    color = get_color(
+                        pseudonym,
+                        dead=death_manager.is_dead(assassin_model),
+                        incompetent=competency_manager.is_inco_at(assassin_model, e.datetime),
+                        is_police=assassin_model.is_police,
+                        is_wanted=wanted_manager.is_player_wanted(assassin_model.identifier, time=e.datetime)
+                    )
+
+                    candidate_pseudonyms.append((assassin_model, pseudonym, color))
```
[the remark this answers is not in the document]
Comment (jsyiek) [reply AU2/plugins/custom_plugins/PageGeneratorPlugin.py:265]: Ah, good spot. The problem looks like it comes from here:

```
pseudonym_index = int(match[0] or 0)
```

File: AU2/plugins/custom_plugins/PageGeneratorPlugin.py [hunk 2]
```diff
@@ -227,6 +232,38 @@ def on_page_generate(self, _) -> List[HTMLComponent]:
...
```
Comment (thenewtonium) [inline AU2/plugins/custom_plugins/PageGeneratorPlugin.py:265]: This still needs to be moved after the iteration over `e.assassins.items()`

**Review by thenewtonium — CHANGES_REQUESTED** [review]

File: AU2/plugins/custom_plugins/PageGeneratorPlugin.py (hunk 2 above)
Comment (thenewtonium) [reply AU2/plugins/custom_plugins/PageGeneratorPlugin.py:265]: (not line +235 ofc)

**Review by thenewtonium — APPROVED** [review]
Lgtm!

**Pull request merged**